sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -201,6 +202,193 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
     }
   }
 
+  test("analyze single partition") {
+    val tableName = "analyzeTable_part"
+
+    def queryStats(ds: String): CatalogStatistics = {
+      val partition =
+        spark.sessionState.catalog.getPartition(TableIdentifier(tableName), Map("ds" -> ds))
+      partition.stats.get
+    }
+
+    withTable(tableName) {
+      sql(s"CREATE TABLE $tableName (key STRING, value STRING) PARTITIONED BY (ds STRING)")
+
+      sql(s"INSERT INTO TABLE $tableName PARTITION (ds='2010-01-01') SELECT * FROM src")
+      sql(
+        s"""
+           |INSERT INTO TABLE $tableName PARTITION (ds='2010-01-02')
+           |SELECT * FROM src
+           |UNION ALL
+           |SELECT * FROM src
+         """.stripMargin)
+      sql(s"INSERT INTO TABLE $tableName PARTITION (ds='2010-01-03') SELECT * FROM src")
+
+      sql(s"ANALYZE TABLE $tableName PARTITION (ds='2010-01-01') COMPUTE STATISTICS").collect()
+
+      sql(s"ANALYZE TABLE $tableName PARTITION (ds='2010-01-02') COMPUTE STATISTICS").collect()
+
+      assert(queryStats("2010-01-01").rowCount.get === 500)
+      assert(queryStats("2010-01-01").sizeInBytes === 5812)
+
+      assert(queryStats("2010-01-02").rowCount.get === 2*500)
+      assert(queryStats("2010-01-02").sizeInBytes === 2*5812)
+    }
+  }
+
+  test("analyze single partition noscan") {
+    val tableName = "analyzeTable_part"
+
+    def queryStats(ds: String): CatalogStatistics = {
+      val partition =
+        spark.sessionState.catalog.getPartition(TableIdentifier(tableName), Map("ds" -> ds))
+      partition.stats.get
+    }
+
+    withTable(tableName) {
+      sql(s"CREATE TABLE $tableName (key STRING, value STRING) PARTITIONED BY (ds STRING)")
+
+      sql(s"INSERT INTO TABLE $tableName PARTITION (ds='2010-01-01') SELECT * FROM src")
+      sql(
+        s"""
+           |INSERT INTO TABLE $tableName PARTITION (ds='2010-01-02')
+           |SELECT * FROM src
+           |UNION ALL
+           |SELECT * FROM src
+         """.stripMargin)
+      sql(s"INSERT INTO TABLE $tableName PARTITION (ds='2010-01-03') SELECT * FROM src")
+
+      sql(s"ANALYZE TABLE $tableName PARTITION (ds='2010-01-01') COMPUTE STATISTICS NOSCAN")
+        .collect()
+
+      sql(s"ANALYZE TABLE $tableName PARTITION (ds='2010-01-02') COMPUTE STATISTICS NOSCAN")
+        .collect()
+
+      assert(queryStats("2010-01-01").rowCount === None)
+      assert(queryStats("2010-01-01").sizeInBytes === 5812)
+
+      assert(queryStats("2010-01-02").rowCount === None)
+      assert(queryStats("2010-01-02").sizeInBytes === 2*5812)
+    }
+  }
+
+  test("analyze a set of partitions") {
```

Comment:
@wzhfy, given that these cases are covered in SparkSqlParserSuite, is it still necessary to cover them again in StatisticsSuite? Partition columns without values are removed at parsing stage so that AnalyzePartitionCommand always receives partition column along with a value.

re: (ds, hr='10') - https://cwiki.apache.org/confluence/display/Hive/StatsDev suggests that it is allowed; this PR supports this syntax.